src/Orleans/Core/GrainClient.cs

```diff
@@ -277,11 +277,22 @@ public static void Uninitialize()
         }
 
         /// <summary>
+        /// Test hook to uninitialize client without cleanup
+        /// </summary>
+        public static void HardKill()
```

Comment:
I agree 100%.  I'm not at all happy with the way this turned out, and think we should revisit this.  I'll open an issue on this.

One reason I made this public was because the OrleansTestingHost does not have access to internals.  I am of the opinion that it should, but wanted to have a discussion on this rather than unilaterally making that decision in this PR.  I'll provide my reasoning for this in the Issue.

A second reason I made this public is that, unlike Silos, we've no means of managing multiple grain clients in our test environments.  We should add this capability, but I was neither willing to bundle those changes into this PR nor withhold these tests until those capabilities were added.